### DISCUSSION TITLE
How do I parse JSON text example from main page?

### DISCUSSION THREAD

**Discussion by githubrandomuser2017** [nlohmann/json#3072] [Q&A] [answered]
The [GitHub main page for this project](https://github.com/nlohmann/json) offers the following JSON example:

```json
{
  "pi": 3.141,
  "happy": true,
  "name": "Niels",
  "nothing": null,
  "answer": {
    "everything": 42
  },
  "list": [1, 0, 2],
  "object": {
    "currency": "USD",
    "value": 42.99
  }
}
```
Suppose I receive that JSON as a string and then instantiate a json object.
```lang-c++
auto j = json::parse(s);
```

Now how do I retrieve keys and values from that JSON?

In particular, how do I get the `list` and `object` items and then read those items? If there isn't already code on the main page to perform this manipulation, can someone add it?

If I have code that looks like this:
```lang-c++
auto list = j["list"];
```
then what exactly is the type of `list`? Is it `vector<int>`?

**Comment by nlohmann** [comment]
The type is again `json` which internally uses a `std::vector<json>` to store arrays. But implicit conversions work as well (see https://github.com/nlohmann/json#conversion-from-stl-containers):

```cpp
std::vector<int> list = j["list"];
```

**Reply by gregmarr** [reply]
>implicit conversions work as well 

Unless you have them disabled.  Also, they're not recommended.

https://github.com/nlohmann/json#implicit-conversions

This is the recommended way to get the data:

    auto list = j["list"].get<std::vector<int>>();

**Comment by nlohmann** [comment]
Do you need further assistance with this issue?

**Comment by githubrandomuser2017** [comment]
Since you put that JSON example at the very top of your project's README, I was hoping that you'd have some example code to read it. Instead, I had to search through a lot of other piecemeal examples to figure out how to do it.

**Reply by nlohmann** [reply]
Is the documentation in https://github.com/nlohmann/json#serialization--deserialization not extensive enough?

**Comment by githubrandomuser2017** [comment, marked as answer]
That section of the documentation is actually _too extensive_. I just want to be able to parse the first JSON example and retrieve the data. Beginners like myself don't want to go through the nuances of N different ways of achieve the task. I was able to figure out how to use the API after three days of reading the documentation, so at this point, I am satisfied with the results.
